### PR TITLE
Roll back the write transaction on schema init if nothing changed

### PR DESCRIPTION
**Pull request by tgoyne**
Eliminates spurious change notifications when opening a Realm flle in multiple processes.

### CODE REVIEW

**Comment by tgoyne** [comment]
Now does way less stuff since all the other bits have been done better in different PRs.

File: Realm/RLMObjectStore.mm [hunk 1]
```diff
@@ -256,9 +209,71 @@ void RLMRealmCreateTables(RLMRealm *realm, RLMSchema *targetSchema, bool updateE
...
+        // apply migration block if provided
+        if (migrating && migrationBlock) {
+            NSError *error = migrationBlock();
+            if (error) {
+                [realm cancelWriteTransaction];
+                return error;
+            }
+        }
+
+        if (oldVersion != newVersion) {
```
Comment (alazier) [inline Realm/RLMObjectStore.mm]: This is the same as ```migrating```. This should be combined with the previous block, with ```if (migrationBlock)``` in a nested code block.

File: Realm/RLMObjectStore.mm [hunk 2]
```diff
@@ -256,9 +209,71 @@ void RLMRealmCreateTables(RLMRealm *realm, RLMSchema *targetSchema, bool updateE
...
+    @catch (NSException *) {
+        [realm cancelWriteTransaction];
+        @throw;
+    }
+
+    return nil;
 }
 
 
+void RLMRealmSetSchema(RLMRealm *realm, RLMSchema *targetSchema, bool verify) {
```
Comment (alazier) [inline Realm/RLMObjectStore.mm]: Can you move this up so it isn't changed in the diff since there are no code changes?

**Comment by alazier** [comment]
Other than the code factoring issues :+1:

**Pull request merged**